addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import datetime
+import uuid
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.addons.l10n_tw_edi_ecpay.utils import EcPayAPI
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # ------------------
+    # Fields declaration
+    # ------------------
+
+    l10n_tw_edi_file_id = fields.Many2one(
+        comodel_name="ir.attachment",
+        compute=lambda self: self._compute_linked_attachment_id("l10n_tw_edi_file_id", "l10n_tw_edi_file"),
+        depends=["l10n_tw_edi_file"],
+        copy=False,
+        readonly=True,
+        export_string_translation=False,
+    )
+    l10n_tw_edi_file = fields.Binary(
+        string="Ecpay JSON File",
+        copy=False,
+        readonly=True,
+        export_string_translation=False,
+    )
+    l10n_tw_edi_ecpay_invoice_id = fields.Char(string="Ecpay Invoice Number", readonly=True, copy=False)
+    l10n_tw_edi_related_number = fields.Char("Related Number", readonly=True, copy=False)
+    # False => Not sent yet.
+    l10n_tw_edi_state = fields.Selection(
+        selection=[("invoiced", "Invoiced"), ("valid", "Valid"), ("invalid", "Invalid")],
+        string="Invoice Status",
+        copy=False,
+        readonly=True,
+        tracking=True,
+    )
+    l10n_tw_edi_ecpay_is_included_tax = fields.Selection(
```

Comment:
As this is used in only one method, you can remove the field and check for the info in the method instead.